lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -180,11 +187,32 @@ def get_d3_problem_grade_distrib(course_id):
                         # Construct label to display for this problem
                         label = "P{0}.{1}.{2}".format(c_subsection, c_unit, c_problem)
 
-                        # Only problems in prob_grade_distrib have had a student submission.
-                        if child.location in prob_grade_distrib:
+                        problem = None
+                        grade_distribution = []
+
+                        try:
+                            problem = client.modules(course_id, child.location)
+                        except NotFoundError:
+                            pass
+
+                        if problem:
```

Comment:
(retyping as the previous version was lost during a code change...)

Alternatively, you could write L190-L199 as:
```python
try:
    problem = client.modules(course_id, child.location)
except NotFoundError:
    grade_distribution = []
else:
    grade_distribution = problem.grade_distribution()
```

It's a bit smaller/compact, but I'll leave that stylistic choice to you.